src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -175,38 +208,59 @@ public ResponseEntity<Void> delete(@PathVariable Long courseId, @PathVariable Lo
-        log.debug("REST request to update TutorialGroup : {}", tutorialGroup);
-        if (tutorialGroup.getId() == null) {
+    public ResponseEntity<TutorialGroup> update(@PathVariable Long courseId, @PathVariable Long tutorialGroupId, @RequestBody @Valid TutorialGroup updatedTutorialGroup) {
+        log.debug("REST request to update TutorialGroup : {}", updatedTutorialGroup);
+        if (updatedTutorialGroup.getId() == null) {
             throw new BadRequestException("A tutorial group cannot be updated without an id");
         }
-        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
-        checkEntityIdMatchesPathIds(tutorialGroup, Optional.of(courseId), Optional.of(tutorialGroupId));
-        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        var oldTutorialGroup = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkEntityIdMatchesPathIds(updatedTutorialGroup, Optional.of(courseId), Optional.of(tutorialGroupId));
+        var responsibleUser = userRepository.getUserWithGroupsAndAuthorities();
+
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, oldTutorialGroup.getCourse(), responsibleUser);
+        trimStringFields(updatedTutorialGroup);
+        if (!oldTutorialGroup.getTitle().equals(updatedTutorialGroup.getTitle())) {
+            checkTitleIsUnique(updatedTutorialGroup);
+        }
 
-        trimStringFields(tutorialGroup);
-        if (!tutorialGroupFromDatabase.getTitle().equals(tutorialGroup.getTitle())) {
-            checkTitleIsUnique(tutorialGroup);
+        // Note: We have to load the teaching assistants from database otherwise languageKey is not defined and email sending fails
+        if (oldTutorialGroup.getTeachingAssistant() == null && updatedTutorialGroup.getTeachingAssistant() != null) {
+            var newTAFromDatabase = userRepository.findOneByLogin(updatedTutorialGroup.getTeachingAssistant().getLogin());
+            newTAFromDatabase.ifPresent(user -> singleUserNotificationService.notifyTutorAboutAssignmentToTutorialGroup(updatedTutorialGroup, user, responsibleUser));
+        }
+        else if (oldTutorialGroup.getTeachingAssistant() != null && updatedTutorialGroup.getTeachingAssistant() == null) {
+            var oldTAFromDatabase = userRepository.findOneByLogin(oldTutorialGroup.getTeachingAssistant().getLogin());
+            oldTAFromDatabase.ifPresent(user -> singleUserNotificationService.notifyTutorAboutUnassignmentFromTutorialGroup(oldTutorialGroup, user, responsibleUser));
+        }
+        else if (oldTutorialGroup.getTeachingAssistant() != null && updatedTutorialGroup.getTeachingAssistant() != null
```

Comment:
Why do you need an extra case? Couldn't you cover this case in two if statements?
```
Case 1: new != null && (old == null || old != new) → send assignment notification to new
Case 2: old != null && (new == null || old != new) → send unassignment notification to old
```

Reply:
I like it! Much cleaner :) Thank you! Changed in 00f5ce4fc7e8717bcec8b07181762180e3041dd5